src/output_writers.jl

```diff
@@ -324,3 +324,72 @@ function time_to_write(clock, out::JLD2OutputWriter)
         return false
     end
 end
+
+mutable struct Checkpointer <: OutputWriter
+                 dir :: String
+              prefix :: String
+    output_frequency :: Int
+end
+
+function Checkpointer(; output_frequency, dir=".", prefix="checkpoint", force=false)
+    mkpath(dir)
+    return Checkpointer(dir, prefix, output_frequency)
+end
+
+function savesubfields!(file, model, name, flds=propertynames(getproperty(model, name)))
+    for f in flds
+        file["$name/$f"] = Array(getproperty(getproperty(model, name), f).data.parent)
+    end
+    return nothing
+end
+
+checkpointed_structs   = [:arch, :boundary_conditions, :grid, :clock, :eos, :constants, :closure]
+checkpointed_fieldsets = [:velocities, :tracers, :G, :Gp]
+
+function write_output(model, c::Checkpointer)
+    @warn "Checkpointer will not save forcing functions, output writers, or diagnostics. They will need to be " *
```

Comment:
Does it make sense to have a warning that is *always* printed? Perhaps is it simply better to document this aspect of the checkpointer?

Reply:
Hmmm, might then be good to just print the warning if there is a non-zero forcing function, or an output writer or diagnostic included.

Then yeah warning may be removed if the checkpointer is well-documented.

Reply:
Here is what I propose:

1. Allow users to set the `modelfields` that are to be checkpoints as an argument to the `Checkpointer` constructor

2. If one of those fields that the user has asked to be saved contains a function, emit a warning.

Reply:
@ali-ramadhan was this resolved?